### DISCUSSION TITLE
haxo-rs issues

### DISCUSSION THREAD

**Discussion by mikenturner** [cardonabits/haxo-hw#24] [Q&A] [answered]
Hi
I'm trying to run the haxo-rs code, but i get the following messages.  Is there something I've missed?
Thanks for any help

`cargo run --release
    Finished release [optimized] target(s) in 1.76s
     Running `target/release/haxo001`
fluidsynth: warning: SDL2 not initialized, SDL2 audio driver won't be usable
fluidsynth: warning: Failed to set thread to high priority
Synth created
Starting haxophone...
fluidsynth: warning: Failed to allocate a synthesis process. (chan=0,key=59)`

the above fluidsynth message occurs everytime a key is pressed

**Comment by jcard0na** [comment]
Hi Mike,

I believe the `SDL2 not initialized error` is benign.  It is raised by fluidsynth, not sure how to disable it  (there is some context here: https://github.com/FluidSynth/fluidsynth/issues/622)

The `Failed to set thread to high priority ` warning only shows up when the program is not run as root.  Should not affect sound, just latency.

The `Failed to allocate a synthesis process` error is more concerning.  I could either be memory pressure or problems with the lower level driver.   Can you play audio files with `aplay`?  Did you enable the audio card (step 5. from [here](https://github.com/cardonabits/haxo-hw/blob/main/docs/development.md#initial-configuration)).

Cheers,

**Reply by mikenturner** [reply]
Hi Javier
Thanks for your responses
1. Ok I can ignore that (i noticed in the link that it used to be an error but is now just a warning)
2. Yes, when I run as root that warning disappears
3. I have enabled the audio card and when I run `aplay -l` I can see the MAX98357A as card 0.  I also see vc4hdmi as card 1.  But I can't seem to play any audio files (though I did get some sounds out of the haxphone yesterday

Mike

**Comment by jcard0na** [comment]
Hi Mike,

Just to compare, I checked the logs on my running system (pasted below).  Here is what I found:

1. I get the same first warning you do.
2. I do not get the second, because I'm [running the service as root via systemd](https://github.com/cardonabits/haxo-hw/blob/main/docs/development.md#installation-and-launch).
3. I do not get the `Failed to allocate a synthesis process`
4. I get a few warnings from fluidsyth regarding driver initialization, but they seem harmless:  the haxophone plays well.
 
```
Aug 14 09:54:01 haxophone haxo001[949]: fluidsynth: warning: SDL2 not initialized, SDL2 audio driver won't be usable
Aug 14 09:54:01 haxophone haxo001[949]: [2023-08-14T08:54:01Z WARN  haxo001::synth] Setting audio.driver in fluidsynth failed
Aug 14 09:54:01 haxophone haxo001[949]: [2023-08-14T08:54:01Z WARN  haxo001::synth] Setting audio.periods in fluidsynth failed
Aug 14 09:54:01 haxophone haxo001[949]: [2023-08-14T08:54:01Z WARN  haxo001::synth] Setting audio.period-size in fluidsynth failed
Aug 14 09:54:01 haxophone haxo001[949]: [2023-08-14T08:54:01Z WARN  haxo001::synth] Setting audio.realtime-prio in fluidsynth failed
```

**Comment by jcard0na** [comment, marked as answer]
BTW, you can find links to pre-compiled images here:  https://github.com/cardonabits/haxo-hw/blob/main/docs/instructions_safety_en.md

**Reply by mikenturner** [reply]
Thanks - I might try those later, but at the moment I'm on holiday (with the haxophone), and can't flash a new image
Mike

**Reply by mikenturner** [reply]
Hi Javier
I used the pre-compiled image and the software seems to work fine.  I had a quick play last night, so it works.  However, the audio jack on the board is not very good, I don't know if anyone else has had issues, but I have to use one finger to hold the jack at a certain angle to get audio (obviously not ideal for playing the haxophone!)
Mike

**Reply by jcard0na** [reply]
Oh, thanks for letting me know.  I have not encounter any problems with the jacks on my boards, but definitely worth looking into it.  Does it happen with all your headsets (assuming you have more than one)?

**Comment by mikenturner** [comment]
Hi
I've got one headset with integral lead and another with separate lead, and both are the same.  I've made a 2nd haxophone HAT as well, so I'll test them on that later
Mike

7 Sept 2023 14:36:50 Javier Cardona ***@***.***>:
